app/src/main/java/com/fady/brain_power_task/MainActivityViewModel.kt

```diff
@@ -1,4 +1,27 @@
 package com.fady.brain_power_task
 
-class MainActivityViewModel {
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.fady.brain_power_task.model.Athlete
+
+class MainActivityViewModel (
+    private val repository: AthleteRepository,
+    application: Application
+) :
+    AndroidViewModel(application) {
```

Comment:
This should extend the `ViewModel()` class not `AndroidViewModel(application)`. It seems like AndroidViewModel is a ViewModel with the application as context, and we usually wouldn't need that.